File: app/src/main/java/com/cxh/mvvmart/di/component/AppComponent.java
package com.cxh.mvvmart.di.component;

import android.content.Context;

import com.cxh.mvvmart.di.moduel.AppModule;
import com.cxh.mvvmart.di.qualifier.ContextLife;

import javax.inject.Singleton;

import dagger.Component;
import okhttp3.OkHttpClient;
import retrofit2.Retrofit;

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/6/7 15:43
 */
@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {

    @ContextLife
    Context getApplication();

    Retrofit getRetrofit();

    OkHttpClient getOkHttpClient();

}

File: app/src/main/java/com/cxh/mvvmart/di/AppComponent.kt
package com.cxh.mvvmart.di

import com.cxh.mvvmart.App

import javax.inject.Singleton

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/6/7 15:43
 */
@Singleton
@Component(modules = arrayOf(AppModule::class, ActivityBindingModule::class, AndroidSupportInjectionModule::class))
interface AppComponent : AndroidInjector<App> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<App>()

}
